tests/test_dashboard.py

```diff
@@ -165,5 +201,61 @@ def test_change_page_to_stats_gur(stats_gur_page: Page):
     ).to_be_visible()
 
 
+def test_change_page_to_weather(weather_page: Page):
+    """Check if the page changes after clicking the 'Weather' button."""
+    page = weather_page
+
+    # Check if the weather page is loaded.
+    expect(page.get_by_role("heading", name="Weather Summary")).to_be_visible()
+
+    # Check if the data has been loaded.
+    expect(page.get_by_text("Weather data fetched")).to_be_visible()
+
+    # Change variable to display.
+    page.locator("div").filter(
+        has_text=re.compile(r"^Wind Speed$")
+    ).first.click()
+    page.get_by_role("option", name="Wind Direction").click()
+
+
+def test_weather_reload_cache(weather_page: Page):
+    """Check if the weather data is reloaded after clicking
+    the refresh button."""
+    page = weather_page
+    page.get_by_test_id("stBaseButton-secondary").click()
+
+    # Check if the data has been reloaded.
+    page.wait_for_timeout(1000)
+    expect(page.get_by_text(
+        "Weather data fetched successfully."
+    )).to_be_visible()
```

Comment:
Changed to wait for the cache to reload then check it has finished. Waiting for a toast was not working